bindings/python/src/session.cpp

```diff
@@ -55,6 +55,7 @@ namespace
 #ifndef TORRENT_DISABLE_DHT
     void add_dht_node(lt::session& s, tuple n)
     {
+        allow_threading_guard guard;
```

Comment:
I don't think it's safe to release the GIL while still accessing python objects (which the following two lines do). Putting it right before the call to add_dht_node() should work though